src/creator.rs

```diff
@@ -97,53 +93,244 @@ impl<H: Hasher> Creator<H> {
         }
     }
 
-    fn check_ready(&self) -> bool {
-        if self.current_round == 0 {
-            return true;
-        }
-        if self.current_round > self.max_round {
-            warn!(target: "AlephBFT-creator", "{:?} Maximum round reached. Not creating another unit.", self.node_ix);
-            return false;
+    async fn wait_until_ready(&mut self, round: Round) {
+
+        let prev_round_index = match round.checked_sub(1) {
+            Some(prev_round) => prev_round as usize,
+            None => return,
```

Comment:
The delay `time::sleep((self.create_lag)(0))` should be applied *before* the 0th round, it's quite important actually -- this is the period when we listen for but not create units.

Reply:
Done